extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef, useState} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useInView} from '../../../src/preact/use-intersect';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const {0: timestamp, 1: setTimestamp} = useState(
+    getFuzzyTimestampValue(props)
```

Comment:
Continuing https://github.com/ampproject/amphtml/pull/26507#discussion_r372624428

Can you upload your amp-bind test? I'm very confident this will not react correctly to changes only to attributes. (If `isInView` changes, it will correctly update)

Reply:
![1c6d84da-b5a9-4c85-8ac3-7ab9afdc0792](https://user-images.githubusercontent.com/10456171/73557207-cd29d580-441e-11ea-91d6-5b7871076a7b.gif)

Reply:
I'm pretty sure this works because when `Timeago` has a prop change, it rerenders. During that rerender, as long as `inView` is truthy, then `setTimestamp` is called (forcing a second render with the updated state).

Here are two musings:
- If we wanted the most efficient version of this code, we could create a `useViewportLayout` hook to only fire on entry instead of also on exit, and then get rid of the if-check and `useState`.
- If we stick with the current `isInView`, its possible a simpler version of the code without checking for `inView` would actually be more efficient since we need to think about whats quicker: double-render on entry or unnecessary render on exit.